### PR TITLE
feat: check sell order availability when buying

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1255

- check sell order availability on pages where it's possible to buy some
- add basic error mapping that can be easily extended later on

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can check the order availability verification here (should work on all pages where it's possible buy sell orders): https://deploy-preview-1313--regen-registry.netlify.app/storefront

File: web-registry/src/config/errors.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { BrokenLinkIcon } from 'web-components/lib/components/icons/BrokenLinkIcon';
+import SellOrderNotFoundIcon from 'web-components/lib/components/icons/SellOrderNotFoundIcon';
+
+export enum ERRORS {
+  DEFAULT,
+  SELL_ORDER_PURCHASED,
+}
+
+export const errorsMapping: Record<
```
Comment (flagrede) [inline web-registry/src/config/errors.tsx:9]: Basic error system that we can extend and use across the app.
The idea is to use the code field later on to match the ledger errors.

**Comment by flagrede** [comment]
@erikalogie the bug you mentioned this morning should be fixed.

**Comment by erikalogie** [comment]
@flagrede yep, this looks good!

I think we should recruit someone else to test this as well since this is a bit of a tricky issue, just to make sure it is working properly. Maybe @wgwz you would have time to try it today?

**Comment by wgwz** [comment]
@erikalogie @flagrede Just wanted to share my test results. In redwood, I found a sell order for kasigau that only had 3 credits available (sell order id 192). upon trying to buy all the credits for the order in two different browsers, i saw the expected result:

brave:
Screen Shot 2022-10-03 at 11 14 59 AM

chrome:
Screen Shot 2022-10-03 at 11 15 18 AM

**Review by wgwz — APPROVED** [review]

**Review by victordelval — COMMENTED** [review]
Super good job!

**Comment by victordelval** [comment]
@flagrede @erika I have only seen that if the user already has the Keplr UI open with the Tx ready, simply waiting to approve the Tx with fees. If at that time the sell order has been modified, the Tx is processed and as a result a false success is obtained (the screen does not show the hash of Tx)

In the image you can see that the user who has purchased is the one on the right side

false_success

**Review by victordelval — APPROVED** [review]

**Comment by flagrede** [comment]
> @flagrede @Erika I have only seen that if the user already has the Keplr UI open with the Tx ready, simply waiting to approve the Tx with fees. If at that time the sell order has been modified, the Tx is processed and as a result a false success is obtained (the screen does not show the hash of Tx)
> 
> In the image you can see that the user who has purchased is the one on the right side
> 
> false_success

Thanks for catching that @victordelval , this is likely a bug will investigate.

**Comment by flagrede** [comment]
@victordelval I was able to fix the issue, thanks again!

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**